packages/vue-flicking/src/Flicking.vue

```diff
@@ -45,64 +68,74 @@ class VueFlicking extends Vue {
     this._initialized = true;
 
     if (this.options.renderOnlyVisible) {
-      // Should update once to update visibles
-      this.$forceUpdate();
+      this.panels = this._getPanelVNodes();
     }
   }
 
-  public beforeMount() {
-    this._fillKeys();
-    this._fillClasses();
+  public beforeDestroy() {
+    this._nativeFlicking.destroy();
   }
 
   public beforeUpdate() {
     this._fillKeys();
-    this._fillClasses();
-  }
 
-  public beforeDestroy() {
-    this._nativeFlicking.destroy();
-  }
+    const slots = this._getSlots();
+    const diffResult = this._slotDiffer.update(slots);
+    const flicking = this._nativeFlicking;
+
+    if (!this.options.renderOnlyVisible || !flicking) {
+      return;
+    }
 
-  public render(h: CreateElement) {
-    const classPrefix = "flicking";
-    const viewportData: VNodeData = {
-      class: {},
-    };
-    const cameraData: VNodeData = {
-      class: {},
-      directives: [{ name: "children-differ", value: this.onUpdate.bind(this) }],
-    };
-    viewportData.class[`${classPrefix}-viewport`] = true;
-    cameraData.class[`${classPrefix}-camera`] = true;
-
-    const panels = this._getPanelElements(h);
-
-    return h(this.viewportTag, viewportData,
-      [h(this.cameraTag, cameraData,
-        panels,
-      )],
-    )
+    const removedPanels = diffResult.removed.reduce((map, idx) => {
+      map[idx] = true;
+      return map;
+    }, {});
+
+    this.panels = [
+      ...flicking.camera.visiblePanels
+        .filter(panel => !removedPanels[panel.index])
+        .map(panel => diffResult.prevList[panel.index]),
+      ...diffResult.added.map(idx => diffResult.list[idx])
+    ];
+
+    this._diffResult = diffResult;
   }
 
-  public onUpdate(diffResult: DiffResult<HTMLElement>) {
+  public updated() {
     const flicking = this._nativeFlicking;
-    const renderer = flicking.renderer as unknown as ExternalRenderer;
+    const renderer = flicking.renderer;
+
+    const slots = this._getSlots();
+    const diffResult = this._diffResult;
 
     diffResult.removed.forEach(idx => {
       renderer.remove(idx, 1);
+      console.log("removed", idx);
```

Comment:
test?

Reply:
Yep, I've removed that in the final commit.